### ISSUE TITLE
Optimizer

### ISSUE THREAD

**Issue by rejasupotaro** [rejasupotaro/master-thesis#10]
## Overview

How do I choose the optimizer for the model?

## Optimizers

[Optimizers](https://keras.io/api/optimizers/)

### Stochastic Gradient Descent (SGD)

Gradient descent  (with momentum) optimizer.

```
tf.keras.optimizers.SGD(
    learning_rate=0.01, momentum=0.0, nesterov=False, name="SGD", **kwargs
)
```

### RMSprop

The gist of RMSprop is to:

- Maintain a moving (discounted) average of the square of gradients
- Divide the gradient by the root of this average

```
tf.keras.optimizers.RMSprop(
    learning_rate=0.001,
    rho=0.9,
    momentum=0.0,
    epsilon=1e-07,
    centered=False,
    name="RMSprop",
    **kwargs
)
```

### Adam: Adaptive moment estimation

Adam = RMSprop + Momentum

According to Kingma et al., 2014, the method is _"computationally efficient, has little memory requirement, invariant to diagonal rescaling of gradients, and is well suited for problems that are large in terms of data/parameters"_.

```
tf.keras.optimizers.Adam(
    learning_rate=0.001,
    beta_1=0.9,
    beta_2=0.999,
    epsilon=1e-07,
    amsgrad=False,
    name="Adam",
    **kwargs
)
```
